wallet/wallet.c

```diff
@@ -1397,33 +1431,39 @@ void wallet_channel_stats_incr_x(struct wallet *w,
 				 struct amount_msat msat)
 {
 	struct db_stmt *stmt;
-	const char *query;
-	/* TODO These would be much better as a switch statement, leaving
-	 * these here for now in order to keep the commit clean. */
-	if (streq(dir, "in") && streq(typ, "offered")) {
+	const char *query = NULL;
+
+	switch (get_state_channel_db(dir, typ)) {
+	case IN_OFFERED:
 		query = SQL("UPDATE channels"
 			    "   SET in_payments_offered = COALESCE(in_payments_offered, 0) + 1"
 			    "     , in_msatoshi_offered = COALESCE(in_msatoshi_offered, 0) + ?"
 			    " WHERE id = ?;");
-	} else if (streq(dir, "in") && streq(typ, "fulfilled")) {
+		break;
+	case IN_FULLFILLED:
 		query = SQL("UPDATE channels"
 			    "   SET in_payments_fulfilled = COALESCE(in_payments_fulfilled, 0) + 1"
 			    "     , in_msatoshi_fulfilled = COALESCE(in_msatoshi_fulfilled, 0) + ?"
 			    " WHERE id = ?;");
-	} else if (streq(dir, "out") && streq(typ, "offered")) {
+		break;
+	case OUT_OFFERED:
 		query = SQL("UPDATE channels"
 			    "   SET out_payments_offered = COALESCE(out_payments_offered, 0) + 1"
 			    "     , out_msatoshi_offered = COALESCE(out_msatoshi_offered, 0) + ?"
 			    " WHERE id = ?;");
-	} else if (streq(dir, "out") && streq(typ, "fulfilled")) {
+		break;
+	case OUT_FULLFILLED:
 		query = SQL("UPDATE channels"
 			    "   SET out_payments_fulfilled = COALESCE(out_payments_fulfilled, 0) + 1"
 			    "     , out_msatoshi_fulfilled = COALESCE(out_msatoshi_fulfilled, 0) + ?"
 			    " WHERE id = ?;");
-	} else {
-		fatal("Unknown stats key %s %s", dir, typ);
+		break;
 	}
 
+	// TODO(vincenzopalazzo): Better here or in the default case of switch stmt?
```

Comment:
this is a perfect place to put this. in general we don't use `default` statements in switches.